### PR TITLE
Enable webview in sandbox renderer

### PR DESCRIPTION
**Pull request by juturu**
webview tag doesn't work in sandbox renderer. This PR enables it.

### CODE REVIEW

File: lib/sandboxed_renderer/init.js [hunk 1]
```diff
@@ -78,6 +78,16 @@ if (window.location.protocol === 'chrome-devtools:') {
   require('../renderer/inspector')
 }
 
+if (binding.guestInstanceId) {
+  process.guestInstanceId = parseInt(binding.guestInstanceId)
+}
+
+if (!process.guestInstanceId && preloadProcess.argv.indexOf('--webview-tag=true') !== -1) {
```
Comment (alexeykuzmin) [inline lib/sandboxed_renderer/init.js]: Can `process .guestInstanceId` be equal to `0` here?

File: spec/api-browser-window-spec.js [hunk 2]
```diff
@@ -1573,6 +1573,19 @@ describe('BrowserWindow module', () => {
         })
         w.loadURL('file://' + path.join(fixtures, 'api', 'preload.html'))
       })
+
+      it.only('webview in sandbox renderer', async () => {
```
Comment (alexeykuzmin) [inline spec/api-browser-window-spec.js]: Please remove the `.only(`.

File: lib/sandboxed_renderer/init.js (hunk 1 above)
Comment (juturu) [reply lib/sandboxed_renderer/init.js]: Guest always start from 1 and go higher.

https://github.com/electron/electron/blob/bcbcb4c6436e84e7f1f2387c2d7581bbdadb5732/lib/browser/guest-view-manager.js#L45-L65

File: spec/api-browser-window-spec.js (hunk 2 above)
Comment (juturu) [reply spec/api-browser-window-spec.js]: why didn't the checks fail on adding .only 🤔

**Comment by tarruda** [comment]
Hari, as we discussed I think it would interesting to enable `sandbox` option inside the `<webview>`. Currently this PR only enables using the `<webview>` tag for a window created with sandbox enabled.

Also, it would probably be a good idea to somehow disable nested `<webview>` so a sandbox webview is not able to create a new `<webview>` with node integration enabled

**Comment by MarshallOfSound** [comment]
> Also, it would probably be a good idea to somehow disable nested <webview> so a sandbox webview is not able to create a new <webview> with node integration enabled

This should be enforced by our child window logic, a webview should never be able to turn on things like `nodeIntegartion` if it's "parent" had them off.

**Comment by juturu** [comment]
>This should be enforced by our child window logic, a webview should never be able to turn on things like nodeIntegartion if it's "parent" had them off.
@MarshallOfSound i'm assuming this is already part of the current child window logic. If yes, are we fine to merge this change?

**Comment by MarshallOfSound** [comment]
Tagged as `semver/breaking`, technically the patch is part bugfix part security fix (enforcing child preference inheritence).

@juturu @alexeykuzmin Care to take a shot: Is this breaking or a bug fix 😄

**Comment by juturu** [comment]
I see this as a bug fix. Any app hitting issues due to this would have a security issue which needs to be fixed immediately.

File: lib/browser/guest-view-manager.js [hunk 3]
```diff
@@ -240,6 +240,22 @@ const attachGuest = function (event, elementInstanceId, guestInstanceId, params)
...
+    ['contextIsolation', true],
+    ['javascript', false],
+    ['nativeWindowOpen', true],
+    ['nodeIntegration', false],
+    ['sandbox', true]
+  ])
+
+  // Inherit certain option values from embedder
+  for (const [name, value] of inheritedWebPreferences) {
+    if (embedder.getLastWebPreferences()[name] === value) {
```
Comment (miniak) [inline lib/browser/guest-view-manager.js]: I would suggest storing the result of getLastWebPreferences()

Comment (miniak) [reply lib/browser/guest-view-manager.js]:
```
const lastWebPreferences = embedder.getLastWebPreferences()
for (const [name, value] of inheritedWebPreferences) {
  if (lastWebPreferences[name] === value) {
     webPreferences[name] = value
  }
}

**Pull request merged**